### ISSUE TITLE
Raise a warning ``using-constant-test`` when casting a generator to a boolean as they are always truthey

### ISSUE THREAD

**Issue by socketpair** [PyCQA/pylint#6909]
### Current problem

Suppose library code:

```python
SOME_DATA = {1:2, 3:4}

def some_function() -> Collection[str]:
    return SOME_DATA.items()
```

and code that use the function:

```python
if some_function():
    do_something()
```

At some point, authors decide to change signature in this way:

```python
def some_function() -> Iterable[str]:
    return ((key, value) for key, value in SOME_DATA.items())
```

Obviously (actually not) the end-user code breaks.

### Desired solution

I propose considering as error all cases where a generator implicitly interpreted as boolean. It will always return `True` even if a generator is exhausted, so this is useless. Error text should contain "to check if value is None use `is not None`".

**Comment by jacobtylerwalls** [comment]
This would be a nice addition to `using-constant-test`.

**Comment by socketpair** [comment]
Actually, I don't understand, will it emit the warning if result of a function previously placed to a variable ?

**Comment by jacobtylerwalls** [comment]
If not, then it's because the name can't be inferred by `astroid` as a `nodes.GeneratorExp`. Could you open an issue there? Another way of putting it: if `n.test.inferred()[0]` below was a `nodes.GeneratorExp`, this would already work (I think).
```python
>>> code = """
... def get_generator():
...     return (x for x in range(0))
... x = get_generator()
... if x:  #@
...   pass
... """
>>> n = astroid.extract_node(code)
>>> n
<If l.5 at 0x102732470>
>>> n.test
<Name.x l.5 at 0x10279c0d0>
>>> n.test.inferred()
[Uninferable]
```
